docs/source/sysadmin_guide/about_dynamic_consensus.rst

```diff
@@ -0,0 +1,247 @@
+
+  .. note::
+
+     For a PoET test network with less than 10 or 12 nodes, disable the
+     defense-in-depth tests (ztest) with these settings:
+
+     .. code-block:: none
+
+        sawtooth.poet.block_claim_delay=1
+        sawtooth.poet.key_block_claim_limit= 100000
+        sawtooth.poet.ztest_minimum_win_count=999999999
+
+
+.. _raft-consensus-label:
+
+Raft Consensus
+==============
+
+:term:`Sawtooth Raft <Raft consensus>` provides a simple consensus algorithm
+that is easy to understand.  It is leader-based, crash fault tolerant, and
+does not fork. For more information, see the
+`Raft documentation <https://sawtooth.hyperledger.org/docs/raft/nightly/master/>`__.
+
+.. note::
+
+   Sawtooth Raft is currently a prototype (not yet released). For more
+   information, see the
+   `sawtooth-raft <https://github.com/hyperledger/sawtooth-raft>`__ repository.
+
+Requirements:
+
+* Each node must install the Raft consensus engine package,
+  ``sawtooth-raft-engine``.
+
+* Each node must run the Raft consensus engine:
+
+  * Service: ``sawtooth-raft-engine.service``
+  * Executable: ``raft-engine``
+
+  For more information, see :doc:`/sysadmin_guide/systemd`. (To start the
+  consensus engine on the command line, see
+  :ref:`start-sawtooth-first-node-label`).
+
+* Specify static peering when starting each validator. Use the ``--peering``
+  option when starting the validator (see :doc:`/cli/sawtooth-validator`) or
+  set the off-chain ``peers`` setting in the ``validator.toml`` configuration
+  file (see :doc:`configuring_sawtooth`).
+
+* Use these on-chain settings for Raft consensus:
+
+  .. code-block:: none
+
+     sawtooth.consensus.algorithm.name=raft
+     sawtooth.consensus.algorithm.version=[VERSION]
+     sawtooth.consensus.raft.peers=[VAL1KEY,VAL2KEY,...,VALnKEY]
+
+  For the version number, see the ``Cargo.toml`` file. Use only the first two
+  digits (for example, `0.1`).
+
+  For ``VALxKEY``, specify the validator public key of each node in the network.
+
+
+.. _devmode-consensus-label:
+
+Devmode Consensus
+=================
+
+Devmode uses a simplified random-leader algorithm. It is intended for testing
+a transaction processor with a single Sawtooth node. Do not use Devmode for a
+production network.
+
+Requirements:
+
+* The node must install the Devmode consensus engine package,
+  ``sawtooth-devmode-engine-rust``.
+
+* The node must run the Devmode consensus engine ``devmode-engine-rust``.
+  For example:
+
+  .. code-block:: console
+
+     $ sudo -u sawtooth devmode-engine-rust -vv --connect tcp://localhost:5050
+
+  For more information, see :doc:`/app_developers_guide/installing_sawtooth`.
+
+* Use these on-chain settings for the Devmode consensus engine:
+
+  .. code-block:: none
+
+     sawtooth.consensus.algorithm.name=Devmode
+     sawtooth.consensus.algorithm.version=0.1
+
+  .. note::
+
+     These settings are optional, because Devmode version 0.1 is the default
```

Comment:
These settings are **not** optional; the genesis validator will fail if they are not specified in 1.2. The backwards compatibility "defaults" are only for the purpose of backwards compatibility. All new networks must explicitly have the name and version settings.